jest-gae/src/main/java/com/searchly/jestgae/JestGaeClient.java

```diff
@@ -0,0 +1,150 @@
+package com.searchly.jestgae;
+
+import com.google.appengine.api.urlfetch.*;
+import com.google.gson.Gson;
+import io.searchbox.action.Action;
+import io.searchbox.client.AbstractJestClient;
+import io.searchbox.client.JestClient;
+import io.searchbox.client.JestResult;
+import io.searchbox.client.JestResultHandler;
+import org.apache.commons.codec.binary.Base64;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Map.Entry;
+
+/**
+ * @author Scopewriter
+ */
+public class JestGaeClient extends AbstractJestClient implements JestClient {
+
+    private final static Logger log = LoggerFactory.getLogger(JestGaeClient.class);
+    private final static String DEFAULT_OK_RESPONSE = "{\"ok\" : true, \"found\" : true}";
+    private final static String DEFAULT_NOK_RESPONSE = "{\"ok\" : false, \"found\" : false}";
+
+    protected final static String REQUEST_CONTENT_TYPE = "application/json; charset=utf-8";
+
+    private URLFetchService urlFetchService = URLFetchServiceFactory.getURLFetchService();
+    private FetchOptions fetchOptions;
+
+    /**
+     * @throws java.io.IOException in case of a problem or the connection was aborted during request,
+     *                     or in case of a problem while reading the response stream
+     */
+    @Override
+    public <T extends JestResult> T execute(Action<T> clientRequest) throws IOException {
+        HTTPRequest request = prepareRequest(clientRequest);
+        HTTPResponse response = urlFetchService.fetch(request);
+        return deserializeResponse(response, clientRequest, request.getMethod());
+    }
+
+    @Override
+    public <T extends JestResult> void executeAsync(final Action<T> clientRequest, final JestResultHandler<T> resultHandler) {
+        throw new UnsupportedOperationException("Jest-gae does not yet support async execution, sorry!");
+    }
+
+    @Override
+    public void shutdownClient() {
+        super.shutdownClient();
+    }
+
+    protected <T extends JestResult> HTTPRequest prepareRequest(final Action<T> clientRequest) throws MalformedURLException {
+        String elasticSearchRestUrl = getRequestURL(getNextServer(), clientRequest.getURI());
+        URL url = new URL(elasticSearchRestUrl);
+        boolean setAuthorizationHeader = false;
+
+        String userInfo = url.getUserInfo();
```

Comment:
Basic authentication should also be supported through the builder as it is done in `io.searchbox.client.config.HttpClientConfig.Builder#defaultCredentials`.